### PR TITLE
Add language labels to code blocks in the documentation

### PR DESCRIPTION
**Pull request by pschiffmann**
Implements #342

My contribution is licensed under the MIT license.

### CODE REVIEW

File: doc/_themes/mantella/static/style.css [hunk 1]
```diff
@@ -212,7 +212,7 @@ main details pre {
   color: #f8f8f8;
 }
 
-main details > div:nth-child(even) pre {
+main details > div:nth-child(3n) pre {
```
Comment (SebastianNiemann) [inline doc/_themes/mantella/static/style.css]: Why was this changed (it is save to assume, that I did not know how it looks with these changes)?

Comment (pschiffmann) [inline doc/_themes/mantella/static/style.css]: [`doc/.preprocessing.py`](https://github.com/Mantella/Mantella/blob/pschiffmann/issue342/doc/.pre_processing.py#L224) inserts the language labels on the same level as the code blocks.

before:
```html
<details>
  <summary>Code example</summary>
  <div class="highlight-c++">...</div>
  <div class="highlight-octave">...</div>
</details>
```

after:
```html
<details>
  <summary>Code example</summary>
  <div class="language"><span>C++</span></div>'
  <div class="highlight-c++">...</div>
  <div class="language"><span>Octave</span></div>'
  <div class="highlight-octave">...</div>
</details>
```

Must be `nth-child(4n - 1)` though.
